components/ProposalCard.tsx

```diff
@@ -0,0 +1,186 @@
+    let against = signatures.filter(x => !x.result).length > contract.threshold;
+    return pro || against;
+  }
+  return (
+    <div className="rounded bg-zinc-800 px-6 py-4">
+      <div>
+        <p className="font-bold text-white md:inline-block">Status: </p>
+        <p className="md:text-md text-sm font-bold text-white md:inline-block">
+          {getState(prop.ui)}
+        </p>
+      </div>
+      {"effective_period" in (contract ?? {}) && (
+        <div>
+          <p className="font-bold text-white md:inline-block">Expires in: </p>
+          <p className="md:text-md text-sm font-bold text-white md:inline-block">
+            {countdown(contract.effective_period, prop.ui.timestamp)}
+          </p>
+        </div>
+      )}
+      <div>
+        <p className="font-bold text-white md:inline-block">Proposed by: </p>
+        <p className="md:text-md text-sm font-bold text-white md:inline-block">
+          {state.aliases[prop.ui.author] || prop.ui.author}
+        </p>
+      </div>
+
+      {("Executed" === prop.ui.status || "Rejected" === prop.ui.status) && (
+        <div>
+          <p className="font-bold text-white md:inline-block">Signed By: </p>
+          <p className="md:text-md text-sm font-bold text-white md:inline-block">
+            [ {[...prop.ui.signatures.keys()].join(", ")} ]
+          </p>
+        </div>
+      )}
+      {"Proposing" === prop.ui.status && (
+        <div>
+          <p className="font-bold text-white md:inline-block">
+            Waiting for signatures from:{" "}
+          </p>
+          <p className="md:text-md text-sm font-bold text-white md:inline-block">
+            [
+            {signers(contract)
+              .filter(x => !!!prop.ui.signatures.find(p => x === p.signer))
+              .map(x => state.aliases[x] || x)
+              .join(", ")}{" "}
+            ]
+          </p>
+        </div>
+      )}
+      <div>
+        <p className="font-bold text-white md:inline-block">Transactions: </p>
+        <p className="md:text-md text-sm font-bold text-white md:inline-block">
+          [
+          {prop.ui.content
+            .map(x => `${renderContent(x, state, address, contract)}`)
+            .join(", ")}{" "}
+          ]
+        </p>
+      </div>
+      <div className="mt-4 flex flex-col md:flex-row">
+        <ContractLoader loading={loading}>
+          {state.address &&
+            signers(contract).includes(state.address) &&
+            signable && (
+              <button
+                type="button"
+                className={
+                  "mx-none mx-auto block w-full self-center justify-self-end rounded bg-primary p-1.5 font-medium text-white hover:bg-red-500 hover:outline-none focus:bg-red-500 md:mx-auto md:w-1/3 md:self-end"
+                }
+                onClick={async e => {
+                  e.preventDefault();
+                  setCloseModal(false);
+                }}
+              >
+                Reject
```

Comment:
This can make in other PR. I think the "Reject" should be replace by "Resolve" when number of signature is greater than or equal to threshold.

Reply:
Yeah you're right, I'll make it in another PR I think, there's already a lot of changes